### ISSUE TITLE
Coverage report generation is broken

### ISSUE THREAD

**Issue by joaosreis** [MinaProtocol/mina#12802]
Running `dune runtest --instrument-with bisect_ppx --force` doesn't run the full test suite, failing at some point of the execution, whereas running the same command without instrumentation runs the whole test suite.

The error shown if the following:
```
File "src/lib/mina_state/snarked_ledger_state.ml", line 479, characters 54-62:
479 |     let ( || ) b1 b2 = Tick.(Run.run_checked Boolean.(b1 || b2))
                                                            ^^^^^^^^
Error: This variant expression is expected to have type
         'a Run.Internal_Basic.Checked.t
       There is no constructor true within type Snarky_backendless__Checked_runner.Simple.Types.Checked.t
```

**Comment by joaosreis** [comment]
I was able to confirm that this is the error preventing the tests from running. Excluding the file from coverage generation, the codebase builds and tests are run.

**Comment by joaosreis** [comment]
Should be fixed by #13005
